### PR TITLE
Vite migration

### PR DESCRIPTION
**Pull request by flomillot**
Based on https://github.com/gridsuite/gridstudy-app/pull/1919 and the latest changes on gridstudy to homogenize the code base.

### CODE REVIEW

File: .browserslistrc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Browsers that we support
+
+[production]
+>0.2% and supports es6-class
+not dead
+not op_mini all
+not chrome < 51
+not safari < 10
```
Comment (Tristan-WorkGH) [inline .browserslistrc:8]: That not the rules in `package.json`
```suggestion
>0.2%
not dead
not op_mini all
```

Comment (Tristan-WorkGH) [reply .browserslistrc:8]: yes, not all fronts are sync, that another work to do, rules aren't in sync with our production env

File: package.json [hunk 2]
```diff
@@ -44,53 +43,22 @@
         "typescript": "5.1.6",
```
Comment (Tristan-WorkGH) [inline package.json:43]: to move in `devDependencies`

File: package.json [hunk 3]
```diff
@@ -44,53 +43,22 @@
...
-        ]
-    },
     "devDependencies": {
+        "@babel/helper-builder-react-jsx": "7.22.10",
         "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
+        "@babel/preset-env": "^7.24.3",
+        "@babel/preset-react": "^7.24.1",
+        "@babel/preset-typescript": "^7.24.1",
         "@types/core-js": "^2.5.8",
+        "@types/eslint": "^7.29.0",
```
Comment (Tristan-WorkGH) [inline package.json:61]: `@types/*` must have major part of version in sync with the lib
```suggestion
        "@types/eslint": "^8.56.9",
```

**Review by Tristan-WorkGH — CHANGES_REQUESTED** [review]

File: .browserslistrc (hunk 1 above)
Comment (flomillot) [reply .browserslistrc:8]: Why we don't use the same ? Is there a problem with the config from gridstudy on gridadmin ?

File: package.json (hunk 2 above)
Comment (flomillot) [reply package.json:43]: Btw it's like that for every projects

File: package.json (hunk 3 above)
Comment (flomillot) [reply package.json:61]: Indeed, so you forgot to it here right ? :p
https://github.com/gridsuite/gridstudy-app/pull/2045/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R82

File: .browserslistrc (hunk 1 above)
Comment (Tristan-WorkGH) [reply .browserslistrc:8]: okay

**Review by Tristan-WorkGH — APPROVED** [review]
LGTM

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**